src/v/kafka/server/replicated_partition.cc

```diff
@@ -394,33 +395,27 @@ ss::future<error_code> replicated_partition::prefix_truncate(
       model::offset_cast(kafka_truncation_offset),
       deadline);
 
-    /// Translate any std::error_codes into proper kafka error codes
-    auto kerr = error_code::unknown_server_error;
-    if (errc.category() == cluster::error_category()) {
-        switch (cluster::errc(errc.value())) {
-        case cluster::errc::success:
-            kerr = error_code::none;
-            break;
-        case cluster::errc::timeout:
-        case cluster::errc::shutting_down:
-            kerr = error_code::request_timed_out;
-            break;
-        case cluster::errc::topic_invalid_config:
-        case cluster::errc::feature_disabled:
+    if (errc.category() == raft::error_category()) {
+        cluster::errc cerrc;
+        switch (raft::errc(errc.value())) {
+        case raft::errc::not_leader:
+            cerrc = cluster::errc::not_leader;
```

Comment:
can we return a particullar error in this case ? without mapping it to the cluster error ?